iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -174,24 +174,16 @@ public override async Task<Message> ReceiveMessageAsync(CancellationToken cancel
             if (Logging.IsEnabled)
                 Logging.Enter(this, cancellationToken, nameof(ReceiveMessageAsync));
 
-            Message message;
-            while (true)
+            try
             {
                 cancellationToken.ThrowIfCancellationRequested();
-
-                using var ctb = new CancellationTokenBundle(_transportSettings.DefaultReceiveTimeout, cancellationToken);
-                message = await _amqpUnit.ReceiveMessageAsync(ctb.Token).ConfigureAwait(false);
-
-                if (message != null)
-                {
-                    break;
-                }
+                return await _amqpUnit.ReceiveMessageAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
We await this call until the cancellation token expires, so this loop is not necessary.

Reply:
No need for the token bundle including the default receive timeout?

Reply:
Since we're using the user supplied cancellation token to signal when this API should be cancelled, we shouldn't be needing the default receive timeout anymore, right?